site/content/docs/guide/plugin/policy-expr.md

```diff
@@ -0,0 +1,193 @@
+modified version of ISO8601. A datetime must include a date in the format
+`<YYYY>-<MM>-<DD>`. An optional time in the format `T<HH>:[MM]:[SS]` will be
+accepted after the date. Decimal fractions of hours and minutes are not allowed;
+use smaller time units instead (e.g. `T10:30` instead of `T10.5`). Decimal
+fractions of seconds are allowed. The timezone is always internally resolved to
+UTC, but you can specify the datetime's original timezone as an an offset from
+UTC by including `+{HH}:[MM]` or `-{HH}:[MM]`.
+
+#### Span
+
+Spans represent a duration of time using the `jiff` [crate] `Span` type. Policy
+expression spans can include weeks, days, hours, minutes, and seconds (including
+decimal fractions of a second). Spans are prefixed with the letter "P" followed
+by optional date units. Time units are separated from date units with the letter
+"T". All date and time units are specified in single case-agnostic letter
+abbreviations after the number. For example, a span of one week, one day, one
+hour, one minute, and one-and-a-tenth seconds would be `P1w1dT1h1m1.1s`.
+
+Although `jiff` day and week spans can be non-uniform depending on timezone
+information, policy expression spans always use uniform 24-hour days and 7-day
+weeks.
+
+### Expressions
+
+#### Arrays
+
+Arrays are vectors of homogeneously-type primitives. This means that all
+elements of an array must be the same type, and that type must be a primitive
+(integer, float, boolean, datetime, span). Arrays cannot contain expression
+types like other arrays, functions, or lambdas. Square brackets represent the
+array boundaries and elements are separated by whitespace. Examples:
+
+ ```
+ [1 1 2 3 5 8]
+ [0.152, -12.482, 0.09]
+ [#t #t #f #t #f]
+ ```
+
+#### Function
+
+Functions are Lisp-like expressions, meaning that they are bounded by
+parentheses, and the function name comes first followed by whitespace-delimited
+operands. Examples:
+
+```
+(add 2 2) // Add two integers
+(min [-3.1, -6.6, 7.8]) // Get the minimum of an array of floats
+```
+
+##### Primitive Function Reference
+
+The standard environment for evaluating policy expressions contains the
+following functions:
+
+| Function | Name | Operand Types | Behavior |
+| ---------| ---- | -------- | -------- |
+| `(gt <A> <B>)`| greater than | non-identifier primitives | evaluate `A > B` |
+| `(lt <A> <B>)`| less than | non-identifier primitives | evaluate `A < B` |
+| `(gte <A> <B>)`| greater than or equal | non-identifier primitives | evaluate `A >= B` |
+| `(lte <A> <B>)`| less than or equal | non-identifier primitives | evaluate `A <= B` |
+| `(eq <A> <B>)` | equal | non-identifier primitives | evaluate `A == B` |
+| `(neq <A> <B>)` | not equal | non-identifier primitives | evaluate `A != B` |
+| `(add <A> <B>)` | add | integers, floats, bools, (datetime and span) | evaluate `A + B` |
+| `(sub <A> <B>)` | subtract | integers, floats, bools, (datetime and span) | evaluate `A - B` |
```

Comment:
Should have some way of saying that you can `add` and `sub` either one datetime and one span _or_ two spans, but _not_ two datetimes to each other.

Reply:
updated